### PR TITLE
docker init

### PR DESCRIPTION
**Pull request by teract10s**

### CODE REVIEW

**Review by mmosii — APPROVED** [review]
LGTM

File: .env [hunk 1]
```diff
@@ -0,0 +1,9 @@
+MYSQLDB_USER=root
+MYSQLDB_ROOT_PASSWORD=1234
+MYSQLDB_DATABASE=book_store
+MYSQLDB_LOCAL_PORT=3308
+MYSQLDB_DOCKER_PORT=3306
+
+SPRING_LOCAL_PORT=8081
+SPRING_DOCKER_PORT=8080
+DEBUG_PORT=5005
```
Comment (liliia-ponomarenko) [inline .env]: how happened that you pushed your .env to GitHub 😭 Please remove it! this file contains  sensitive  information

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+#Final stage
+FROM openjdk:17-jdk-slim
+WORKDIR application
+COPY --from=builder application/dependencies/ ./
+COPY --from=builder application/spring-boot-loader/ ./
+COPY --from=builder application/snapshot-dependencies/ ./
+COPY --from=builder application/application/ ./
+ENTRYPOINT ["java", "org.springframework.boot.loader.JarLauncher"]
+EXPOSE 8080
```
Comment (liliia-ponomarenko) [inline Dockerfile]: add one empty line

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    restart: on-failure
+    depends_on:
+      - book-store-db
+    env_file: ./.env
+    ports:
+      - $SPRING_LOCAL_PORT:$SPRING_DOCKER_PORT
+      - $DEBUG_PORT:$DEBUG_PORT
+    environment:
+      SPRING_APPLICATION_JSON: '{
+        "spring.datasource.url" : "jdbc:mysql://book-store-db:3306/book_store?serverTimezone=UTC",
```
Comment (liliia-ponomarenko) [inline docker-compose.yml]: use env variables  instead `3306/book_store`

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        "spring.datasource.username" : "$MYSQLDB_USER",
+        "spring.datasource.password" : "$MYSQLDB_ROOT_PASSWORD"
```
Comment (liliia-ponomarenko) [inline docker-compose.yml]: Let’s  add drier property

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -1,10 +1,13 @@
...
+spring.datasource.url=jdbc:mysql://book-store-db:3306/book_store?serverTimezone=UTC
 spring.datasource.username=root
 spring.datasource.password=1234
 spring.datasource.driverClassName=com.mysql.cj.jdbc.Driver
+
 spring.jpa.hibernate.ddl-auto=validate
 spring.jpa.show-sql=true
 spring.jpa.open-in-view=false
+
 server.servlet.context-path=/api
```
Comment (liliia-ponomarenko) [inline src/main/resources/application.properties]: think which env variable  you can use in this file

File: .env (hunk 1 above)
Comment (liliia-ponomarenko) [reply .env]: you can add `.env.sample` but it file without values (for now it’s an optional comment)

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s  fix a few mistakes ;)

File: .gitignore [hunk 6]
```diff
@@ -31,3 +32,5 @@ build/
 
 ### VS Code ###
 .vscode/
+
+
```
Comment (liliia-ponomarenko) [inline .gitignore:36]:
```suggestion
```

**Review by liliia-ponomarenko — APPROVED** [review]
Great! 😊

**Pull request merged**